### PR TITLE
vdi_image_convert: new case of vdi_image_convert

### PR DESCRIPTION
**Pull request by hellohellenmao**
1. Convert images between raw and vdi

Signed-off-by: Tingting Mao <timao@redhat.com>

ID: 2072406

### CODE REVIEW

**Comment by hellohellenmao** [comment]
(1/2) Host_RHEL.m9.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.vdi_image_convert.to_raw.q35: STARTED
- (1/2) Host_RHEL.m9.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.vdi_image_convert.to_raw.q35: PASS (3.14 s)
 (2/2) Host_RHEL.m9.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.vdi_image_convert.to_vdi.q35: STARTED
\ (2/2) Host_RHEL.m9.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.vdi_image_convert.to_vdi.q35: PASS (3.22 s)

**Comment by hellohellenmao** [comment]
@zhencliu Could you please help to take a review? Thanks.

File: qemu/tests/cfg/vdi_image_convert.cfg [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    write_size = 512M
+    variants:
+        - to_raw:
+            images = "vdi_src"
+            image_name_vdi_src = "images/vdi_src"
+            image_size_vdi_src = 1G
+            image_format_vdi_src = vdi
+            convert_target = target
+            image_name_target = "images/vdi_to_raw"
+            image_format_target = raw
```
Comment (zhencliu) [inline qemu/tests/cfg/vdi_image_convert.cfg:18]: a little bit of dup here for src and target, what about:

images = 'src'
image_name_src = 'images/src'
image_size_src = 1G
convert_target = target
image_name_target = "images/target"
variants:
-  to_raw:
   image_format_src = vdi
   image_format_target = raw
-  to_vdi:
   image_format_src = raw
   image_format_target = vdi

File: qemu/tests/vdi_image_convert.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    :param params: Dictionary with the test parameters.
+    :param env: Dictionary with test environment.
+    """
+
+    def _qemu_io(img, cmd):
+        """Run qemu-io cmd to a given img."""
+        try:
+            QemuIOSystem(test, params, img.image_filename).cmd_output(cmd, 120)
+        except process.CmdError:
+            test.fail("qemu-io to '%s' failed." % img.image_filename)
```
Comment (zhencliu) [inline qemu/tests/vdi_image_convert.py:22]: test.error here? This is not a test scenario of this case.

File: qemu/tests/cfg/vdi_image_convert.cfg (hunk 1 above)
Comment (hellohellenmao) [reply qemu/tests/cfg/vdi_image_convert.cfg]: Yes. Updated, thanks.

File: qemu/tests/vdi_image_convert.py (hunk 2 above)
Comment (hellohellenmao) [reply qemu/tests/vdi_image_convert.py]: Updated, thanks.

**Comment by hellohellenmao** [comment]
(1/2) Host_RHEL.m9.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.vdi_image_convert.to_raw.q35: STARTED
- (1/2) Host_RHEL.m9.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.vdi_image_convert.to_raw.q35: PASS (3.18 s)
 (2/2) Host_RHEL.m9.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.vdi_image_convert.to_vdi.q35: STARTED
\ (2/2) Host_RHEL.m9.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.vdi_image_convert.to_vdi.q35: PASS (3.23 s)

**Comment by hellohellenmao** [comment]
@zhencliu Could you please help to review again? Thanks.

**Comment by zhencliu** [comment]
LGTM

**Pull request merged**